TFT/src/User/API/Gcode/mygcodefs.c

```diff
@@ -112,14 +112,14 @@ bool scanPrintFilesGcodeFs(void)
         else
           Pstr_tmp++;
 
-        infoFile.longFile[infoFile.fileCount] = malloc(strlen(Pstr_tmp) + 2);  // plus one extra byte for filename extension check
+        infoFile.longFile[infoFile.fileCount] = malloc(strlen(Pstr_tmp) + 2);  // "+ 2": space for terminating null character and the flag for filename extension check
         if (infoFile.longFile[infoFile.fileCount] == NULL)
         {
           clearRequestCommandInfo();
           break;
         }
-        strcpy(infoFile.longFile[infoFile.fileCount], Pstr_tmp);
-        infoFile.longFile[infoFile.fileCount][strlen(Pstr_tmp) + 1] = 0;  // set to 0 the extra byte for filename extension check
+        // copy file name and set the flag for filename extension check
+        strncpy(infoFile.longFile[infoFile.fileCount], Pstr_tmp, strlen(Pstr_tmp) + 2);  // "+2":  terminating null character and the flag for filename extension check
```

Comment:
> this is not correct. you are not setting the extension flag.

Than how do you explain that this code in "myfatfs.c"
```
      infoFile.file[infoFile.fileCount][len] = 1;  // make sure file extension flag is NOT set 
      strncpy(infoFile.file[infoFile.fileCount], finfo.fname, len + 1);
```
and this code in "vfs.c"
```
char * hideExtension(char * filename)
{
  char * extPos = isSupportedFile(filename);

  if (extPos != NULL)  // if filename provides a supported filename extension
  {
    if (filename[strlen(filename) + 1] == '\0')
        filename[extPos - filename] = '*';  // temporarily hide filename extension
  }
  
  return filename;
}
```

resulted this: 
![TerminalChar](https://user-images.githubusercontent.com/50251547/147093179-ae2dccf8-b2bb-429a-b4b6-44bdcd04c9b6.jpg)

Magic? Luck? I don't think so...
You probably do not understand how `strncpy` works. A quick search on Google yields this:
"The C library function `char *strncpy(char *dest, const char *src, size_t n)` copies up to n characters from the string pointed to, by src to dest. **In a case where the length of src is less than that of n, the remainder of dest will be padded with null bytes.**"

You can also look up `strncpy`  in the library itself.



> it is very dangerous to try to copy more data than Pstr_tmp really have (the max number of byte to copy must always be <= strlen(Pstr_tmp)

It's not how `strncpy` works, since "Pstr_tmp" is the source, not the destination.